coil-base/src/main/java/coil/map/HttpStringMapper.kt

```diff
@@ -0,0 +1,10 @@
+package coil.map
+
+import okhttp3.HttpUrl
+
+internal class HttpStringMapper : Mapper<String, HttpUrl> {
+
+    override fun handles(data: String): Boolean = HttpUrl.parse(data) != null
```

Comment:
I don't have benchmarks to verify this, but I believe this adds a non-zero amount of overhead to every String URL request since we end up calling `HttpUrl.get` twice. Mappers are also called on the main thread so they need to be fast + non-blocking.